### DISCUSSION TITLE
Glyphs are hard to read when used with text. Thoughts on setting Glyph sizes?

### DISCUSSION THREAD

**Discussion by TheEmbracedOne** [Orange-Panda/Rewired-Glyphs#15] [Q&A] [answered]
This is potentially feature request territory - is there any way currently to set glyph sizes? If not, what do you think about the stuff below?

When I use a Glyph on its own, it's easy to adjust its size, but sometimes, for example on onscreen displays or tutorials, I display glyphs and texts in one TMProUGUI component. This is great because I don't need to create a multi-gameobject text display (ie text - image - text) and force it to look like it's one string via a Horizontal Layout Group, like I used to before I started using Rewired Glyphs, instead I can do `Hold <Glyph ResetView type=mouse> or <Glyph ResetView type=keyboard>` and it just displays what I need, glyphs and texts, and with my game having a font size slider, this is really handy so I get to have less Layout Groups to adjust the spacing of on bigger or smaller font sizes.

However, glyphs are just too hard to read in most cases. Here's a view of a font size 35 setup of example prompts:
![Unity_X53W9U2Rgv](https://github.com/Orange-Panda/Rewired-Glyphs/assets/16197906/4ad5ce37-a537-45ea-ae5a-cdcc64577db2)

and for this example to illustrate my point, I split them into separate strings, and use a Horizontal Layout Group to display them as if they were one string, adjusting glyphs to 80 font size. The text is still font size 35:
![image](https://github.com/Orange-Panda/Rewired-Glyphs/assets/16197906/a822254f-3491-4a9a-aae7-3ddf45aa4b6f)

I did a lot of tests, updated my glyph assets (I now use a sprite sheet where all assets are 220 x 220 pixels, and I adjusted them so they are relative in size to one another instead of just taking up all the space, which helps:
![image](https://github.com/Orange-Panda/Rewired-Glyphs/assets/16197906/28ef5f87-33af-4c87-9df4-d617dd03793f)

I'm not sure what the best solution to this would be, if it's even possible, but it would be nice if there was a way to "tell" the glyph via the GlyphRichTextFormatter, to be "x" bigger in font size than it was to begin with. 
If this was possible, in my above example I use font size 35 for the texts, but glyphs would use 80 font size, I'd enter "45" as "x".

Perhaps this is not possible since the text and glyphs act as one TMProUGUI component (or maybe somehow the submesh could be "told" to use a different font size?) Or perhaps the complexity of this is not worth it and I should just use Horizontal Layout Groups and separate TMProUGUI objects for the glyphs that they can be then adjusted manually.

But before I make any drastic changes, I'd love to hear what you think!

**Comment by Orange-Panda** [comment]
The sizing of inline sprite can be configured on the generated TMP_SpriteAsset. This does impact the sizing everywhere it is used but that may be acceptable if this is a general problem with sprite sizing. You may also want to consider using TMP's `<size>` [rich text tag](http://digitalnativestudios.com/textmeshpro/docs/rich-text/) around your glyph tag to adjust its font size in that specific context.

As far as features that could be implemented on my end I could perhaps add a field to the generator that sets the sizing of the SpriteAsset when generated but this would be a convenience feature and not something required for you to make adjustments on your end.

**Reply by TheEmbracedOne** [reply]
Oh, I have not considered the <size> tag, and didn't know you can tweak sizes via the generated TMP_SpriteAsset, I'll have a look at both of these today.

The <size> tag might not be optimal for me because I have a font size slider in the settings, I'll have to do some testing to see how it looks in situ! The spritesheet tweaking sounds like a potentially great "global" solution for my project though.

I'll have a look at both things and report back my findings. Thank you!

**Reply by Orange-Panda** [reply, marked as answer]
You may still find the size tag useful with a font size preference since you can use it as a % of the base font size such as `<size=150%>`. No matter how you decide to approach it though you've got lots options to experiment with and find what's best for you and your project

**Reply by TheEmbracedOne** [reply]
Thank you, this helped a lot. I was able to adjust the OY and SF values of the generated sprite sheet with my desired values and was able to get the sprites to align nicely, too.
